src/SFML/Window/DRM/DRMContext.cpp

```diff
@@ -0,0 +1,594 @@
+//    and must not be misrepresented as being the original software.
+//
+// 3. This notice may not be removed or altered from any source distribution.
+//
+////////////////////////////////////////////////////////////
+
+////////////////////////////////////////////////////////////
+// Headers
+////////////////////////////////////////////////////////////
+#include <SFML/Window/DRM/DRMContext.hpp>
+#include <SFML/Window/DRM/WindowImplDRM.hpp>
+#include <SFML/OpenGL.hpp>
+#include <SFML/System/Err.hpp>
+#include <SFML/System/Sleep.hpp>
+#include <cerrno>
+#include <cstdlib>
+#include <cstring>
+#include <poll.h>
+#include <unistd.h>
+
+namespace
+{
+    bool initialized = false;
+    drm drmNode;
+    drmEventContext drmEventCtx;
+    pollfd pollFD;
+    gbm_device* gbmDevice = NULL;
+    int contextCount = 0;
+    EGLDisplay display = EGL_NO_DISPLAY;
+    int waitingForFlip = 0;
+
+    static void pageFlipHandler(int fd, unsigned int frame,
+        unsigned int sec, unsigned int usec, void* data)
+    {
+        // suppress unused param warning
+        (void)fd, (void)frame, (void)sec, (void)usec;
+
+        int* temp = static_cast<int*>(data);
+        *temp = 0;
+    }
+
+    static bool waitForFlip(int timeout)
+    {
+        while (waitingForFlip)
+        {
+            pollFD.revents = 0;
+
+            if (poll(&pollFD, 1, timeout) < 0)
+                return false;
+
+            if (pollFD.revents & (POLLHUP | POLLERR))
+                return false;
+
+            if (pollFD.revents & POLLIN)
+            {
+                drmHandleEvent(drmNode.fd, &drmEventCtx);
+            }
+            else
+            {
+                return false;
+            }
+        }
+        return true;
+    }
+
+    void cleanup()
+    {
+        if (!initialized)
+            return;
+
+        drmModeSetCrtc(drmNode.fd,
+                       drmNode.original_crtc->crtc_id,
+                       drmNode.original_crtc->buffer_id,
+                       drmNode.original_crtc->x,
+                       drmNode.original_crtc->y,
+                       &drmNode.connector_id,
+                       1,
+                       &drmNode.original_crtc->mode);
+
+        drmModeFreeConnector(drmNode.saved_connector);
+        drmModeFreeEncoder(drmNode.saved_encoder);
+        drmModeFreeCrtc(drmNode.original_crtc);
+
+        eglTerminate(display);
+        display = EGL_NO_DISPLAY;
+
+        gbm_device_destroy(gbmDevice);
+        gbmDevice = NULL;
+
+        close(drmNode.fd);
+
+        drmNode.fd = -1;
+        drmNode.mode = 0;
+
+        std::memset(&pollFD, 0, sizeof(pollfd));
+        std::memset(&drmEventCtx, 0, sizeof(drmEventContext));
+
+        waitingForFlip = 0;
+
+        initialized = false;
+    }
+
+    void checkInit()
+    {
+        if (initialized)
+            return;
+
+        // device: Use environment variable "SFML_DRM_DEVICE" (or NULL if not set)
+        char* deviceString = getenv("SFML_DRM_DEVICE");
```

Comment:
`getenv` -> `std::getenv` here and in at least a few other spots in this function